core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -4202,67 +4205,54 @@ private static class LogicalNotImplementor extends AbstractRexCallImplementor {
    * appropriate base (i.e. base e for LN).
    */
   private static class LogImplementor extends AbstractRexCallImplementor {
-    LogImplementor() {
+    private final SqlLibrary library;
+    LogImplementor(SqlLibrary library) {
       super("log", NullPolicy.STRICT, true);
+      this.library = library;
     }
 
     @Override Expression implementSafe(final RexToLixTranslator translator,
         final RexCall call, final List<Expression> argValueList) {
-      return Expressions.call(BuiltInMethod.LOG.method, args(call, argValueList));
-    }
-
-    private static List<Expression> args(RexCall call,
-        List<Expression> argValueList) {
-      Expression operand0 = argValueList.get(0);
-      final Expressions.FluentList<Expression> list = Expressions.list(operand0);
-      switch (call.getOperator().getName()) {
-      case "LOG":
-        if (argValueList.size() == 2) {
-          return list.append(argValueList.get(1)).append(Expressions.constant(0));
-        }
-        // fall through
-      case "LN":
-        return list.append(Expressions.constant(Math.exp(1))).append(Expressions.constant(0));
-      case "LOG10":
-        return list.append(Expressions.constant(BigDecimal.TEN)).append(Expressions.constant(0));
-      default:
-        throw new AssertionError("Operator not found: " + call.getOperator());
+      if (library == SqlLibrary.MYSQL) {
+        return Expressions.call(BuiltInMethod.LOG.method, args(call, argValueList, 1));
+      } else if (library == SqlLibrary.POSTGRESQL) {
+        return Expressions.call(BuiltInMethod.LOG.method, args(call, argValueList, 2));
+      } else {
+        return Expressions.call(BuiltInMethod.LOG.method, args(call, argValueList, 0));
       }
     }
-  }
-
-  /** Implementor for the {@code LN}, {@code LOG}, {@code LOG2} and {@code LOG10} operators
-   *  on Mysql and Spark library
-   *
-   * <p>Handles all logarithm functions using log rules to determine the
-   * appropriate base (i.e. base e for LN).
-   */
-  private static class LogMysqlImplementor extends AbstractRexCallImplementor {
-    LogMysqlImplementor() {
-      super("log", NullPolicy.STRICT, true);
-    }
-
-    @Override Expression implementSafe(final RexToLixTranslator translator,
-        final RexCall call, final List<Expression> argValueList) {
-      return Expressions.call(BuiltInMethod.LOG.method, args(call, argValueList));
-    }
 
     private static List<Expression> args(RexCall call,
-        List<Expression> argValueList) {
-      Expression operand0 = argValueList.get(0);
+        List<Expression> argValueList, int flag) {
+      Expression operand0;
+      Expression operand1;
+      if (flag == 0 && argValueList.size() == 2) {
```

Comment:
I don't know if this part of the code can be optimized, but it seems to be the only way to match it